psycopg/psycopg/types/geometry.py

```diff
@@ -0,0 +1,86 @@
+"""
+Adapters for PostGIS geometries
+"""
+
+from typing import Optional, Type
+
+from .. import postgres
+from ..abc import AdaptContext
+from ..adapt import Dumper, Loader
+from ..pq import Format
+from .._typeinfo import TypeInfo
+
+
+try:
+    import shapely.wkb as wkb
+    from shapely.geometry.base import BaseGeometry
+
+except ImportError:
+    raise ImportError(
+        "The module psycopg.types.geometry requires the package 'Shapely'"
+        " to be installed"
+    )
+
+
+class GeometryBinaryLoader(Loader):
+    format = Format.BINARY
+
+    def load(self, data: bytes) -> "BaseGeometry":
+        return wkb.loads(data)
+
+
+class GeometryLoader(Loader):
+    format = Format.TEXT
+
+    def load(self, data: bytes) -> "BaseGeometry":
+        # it's a hex string in binary
+        return wkb.loads(data.decode(), hex=True)
```

Comment:
Likewise, if a memoryview ends in this function (as it normally does), it will not have a decode() method available.

```diff
 class GeometryLoader(Loader):
     format = Format.TEXT
 
-    def load(self, data: bytes) -> "BaseGeometry":
+    def load(self, data: Buffer) -> "BaseGeometry":
+        if isinstance(data, memoryview):
+            data = bytes(data)
         # it's a hex string in binary
         return wkb.loads(data.decode(), hex=True)
```